test/Riok.Mapperly.IntegrationTests/Dto/TestObjectDto.cs

```diff
@@ -53,6 +53,8 @@ public TestObjectDto(int ctorValue, int unknownValue = 10, int ctorValue2 = 100)
 
         public TestObjectNestedDto[]? NullableReadOnlyObjectCollection { get; set; }
 
+        public Stack<int> EnsureCapacityValue { get; set; } = new();
```

Comment:
Is this separate test property really required? What additional test case does it test?

Reply:
`Enumerable.TryGetNonEnumeratedCount` was added in .Net 6+. If a lower version is used a statement using pattern matching is created (`EnsureCapacityBuilderIsType`). I took advantage of the .Net Framework 4.8 integration test to ensure that this works.

I could probably remove `EnsureCapacityBuilderIsType` as its never used in .Net 6+

Reply:
Ah now I got it, this tests a mapping `IEnumerable<string>` => `Stack<int>` (which uses the type pattern matching / TryGetNonEnumeratedCount respectively), `Stack<string>` => `Stack<int>` tests the case for which the source count is available directly.
I'd keep `EnsureCapacityIsType` for now, as Mapperly tries to support .NET Framework as good as possible.